src/components/AddComponent/GitAuthorization.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import {
+  Button,
+  ButtonVariant,
+  Popover,
+  Text,
+  TextContent,
+  TextVariants,
+} from '@patternfly/react-core';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons/dist/js/icons/outlined-question-circle-icon';
+import { useField } from 'formik';
+import { useActiveNamespace } from '../../hooks/useActiveNamespace';
+import { LoadingInline } from '../../shared/components/status-box/StatusBox';
+import { initiateAccessTokenBinding } from '../../utils/create-utils';
+import { useAccessTokenBindingAuth } from './utils';
+
+export const GitAuthorization: React.FC = () => {
```

Comment:
If someone initiates oauth flow, but then closes the window. How do they re-establish the auth flow with the existing request binding?
It doesn't seem like that happens now (although I haven't tried this PR).
Also what about cleaning up `SPIAccessTokenBinding` that are no longer relevant? for example if the user changes the git repo URL after going through the auth flow, or leaving the page without creating the component.

Reply:
Right now the url and access binding context is only stored in formik state, and we'd lose all of it if the window is closed.
Should we save this context in local storage maybe?

Also note that users don't have edit/delete access to `SPIAccessTokenBinding`s, so not sure how we'd clean it up after creation.

Reply:
Updated.

The hook will now look for an existing `SPIAccessTokenBinding` for the ULR before creating a new one and use that instead if available.
And the created `SPIAccessTokenBinding` will be deleted on unmount.

Reply:
Deleting on unmount would never have worked because unmount could occur whether they completed the form or not. The idea was only to delete if they left the flow without completing it.

As per another comment, since we are reusing existing bindings, let's keep them around for now and revisit this later.